src/main/java/com/thebois/models/world/structures/AbstractStructure.java

```diff
@@ -68,10 +70,14 @@ private int countAllNeededItems() {
     @Override
     public boolean tryDeliverItem(final IItem deliveredItem) {
         final Collection<ItemType> neededItems = getNeededItems();
+        boolean successfulDelivery = false;
         if (neededItems.contains(deliveredItem.getType())) {
-            return deliveredItems.tryAdd(deliveredItem);
+            successfulDelivery = deliveredItems.tryAdd(deliveredItem);
         }
-        return false;
+        if (isCompleted()) {
+            postSpawnPawnsEvent(getPosition());
+        }
```

Comment:
This will make all structures spawn pawns, which isn't really what we want, right?